ksqldb-engine/src/main/java/io/confluent/ksql/query/PullQueryQueue.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.query;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.physical.pull.PullQueryRow;
+import io.confluent.ksql.util.KeyValue;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class PullQueryQueue implements BlockingRowQueue {
+
+  public static final int BLOCKING_QUEUE_CAPACITY = 50;
+  private static final Function<PullQueryRow, KeyValue<List<?>, GenericRow>>
+      PQ_ROW_TO_KEY_VALUE = row -> KeyValue.keyValue(null, row.getGenericRow());
+
+  private final BlockingQueue<PullQueryRow> rowQueue;
+  private final int offerTimeoutMs;
+  private volatile boolean closed = false;
+
+  private LimitHandler limitHandler;
+  private LimitQueueCallback callback;
```

Comment:
```suggestion
  private LimitQueueCallback onLimitExceededCallback;
```

Reply:
I changed this to `Runnable queuedCallback` since that's all it does now.